src/components/oscal-utils/OSCALRestUtils.js

```diff
@@ -184,22 +186,51 @@ export function updateSspControlImplementationImplementedRequirementStatementByC
 ) {
   const partialRestImplementedRequirement = deepClone(implementedRequirement);
 
-  // Find our statementByComponent and update the description
-  const statementByComponent = partialRestImplementedRequirement.statements
-    .find((element) => element["statement-id"] === statementId)
-    ["by-components"].find(
-      (element) => element["component-uuid"] === componentId
-    );
+  // Find our statement
+  let statement = partialRestImplementedRequirement?.statements?.find(
+    (element) => element["statement-id"] === statementId
+  );
+
+  // If our statement doesn't exist, create and add it
+  if (!statement) {
+    statement = {
+      "statement-id": props.statementId,
+      uuid: uuidv4(),
+      "by-components": [],
+    };
+    if (!partialRestImplementedRequirement.statements) {
+      partialRestImplementedRequirement.statements = [];
+    }
+    partialRestImplementedRequirement.statements.push(statement);
+  }
+
+  // Find our statementByComponent
+  let statementByComponent = statement["by-components"].find(
+    (element) => element["component-uuid"] === componentId
+  );
+  // If our statementByComponent doesn't exit, create and add it
+  if (!statementByComponent) {
+    statementByComponent = {
+      "component-uuid": componentId,
+      uuid: uuidv4()
+    };
+    if (!statement["by-components"]) {
+      statement["by-components"] = [];
+    }
+    statement["by-components"].push(statementByComponent);
+  }
```

Comment:
If `statement["by-components"]` is null-like (a case we check for by initializing to `[]` later), the first `find` will throw an error.

```suggestion
  // Find our statementByComponent
  let statementByComponent = statement["by-components"]?.find(
    (element) => element["component-uuid"] === componentId
  );
  // If our statementByComponent doesn't exit, create and add it
  if (!statementByComponent) {
    statementByComponent = {
      "component-uuid": componentId,
      uuid: uuidv4()
    };
    statement["by-components"] ??= [];
    statement["by-components"].push(statementByComponent);
  }
```

Reply:
Done.